helper/ssh/key_pair.go

```diff
@@ -139,7 +139,7 @@ func (o *defaultKeyPairBuilder) Build() (KeyPair, error) {
 		return o.newEcdsaKeyPair()
 	}
 
-	return nil, fmt.Errorf("Unsupported keypair type: %s", o.kind.String())
+	return defaultKeyPair{}, fmt.Errorf("Unsupported keypair type: %s", o.kind.String())
```

Comment:
That is a good point. We waffled on this a bit, and decided having a default would be nice to avoid a crash. Ultimately, you are correct - if the caller is ignoring the error, than it is better to crash and expose the deficiency. Also, it looks like we forgot to return the default in all the cases anyways.

Reply:
@rickard-von-essen, We ended up using a struct to represent the key pair instead of a pointer or an interface (per review feedback from @azr). I am going to resolve this, but you are welcome to write additional feedback if you like.